scripts/gff_metaparser/genmetaconf/seqregionconf.py

```diff
@@ -26,364 +26,388 @@
 from typing import Dict, List, Optional
 
-from Bio import SeqIO # type: ignore
+from Bio import SeqIO  # type: ignore
+
 
 @dataclass
 class SeqRegionSyn:
-  name: str
-  source: str
+    name: str
+    source: str
+
 
 @dataclass
 class KaryotypeBand:
-  name: str
-  start: int
-  end: int
-  stain: Optional[str]
-  structure: Optional[str]
+    name: str
+    start: int
+    end: int
+    stain: Optional[str]
+    structure: Optional[str]
+
 
 @dataclass
 class SeqRegion:
-  name: str
-  length: int
-  coord_system_level: str = "contig"
-  location: Optional[str] = None
-  circular : Optional[bool] = None
-  codon_table : Optional[int] = None
-  synonyms: List[SeqRegionSyn] = field(default_factory=list)
-  karyotype_bands: List[KaryotypeBand] = field(default_factory=list)
-  _rank: Optional[int] = None
+    name: str
+    length: int
+    coord_system_level: str = "contig"
+    location: Optional[str] = None
+    circular: Optional[bool] = None
+    codon_table: Optional[int] = None
+    synonyms: List[SeqRegionSyn] = field(default_factory=list)
+    karyotype_bands: List[KaryotypeBand] = field(default_factory=list)
+    _rank: Optional[int] = None
+
 
 def no_nulls_dict(x):
-  return {k:v for k, v in x if v is not None}
+    return {k: v for k, v in x if v is not None}
+
 
 class SeqRegionConf:
-  def __init__(self,
-               fasta_file: Optional[str],
-               asm_rep_file: Optional[str],
-               seq_region_raw: Optional[str],
-               seq_region_genbank: Optional[str],
-               seq_region_syns: Optional[str],
-               meta: Optional[dict],
-               syns_src: str = "GenBank",
-               default_genetic_code = 1,
-               default_circular = False
-               ):
-    seq_region_factory = lambda:SeqRegion("",0) 
-    if default_genetic_code != 1 or default_circular != False:
-      seq_region_factory = lambda:SeqRegion("",0, codon_table = default_genetic_code, circular = default_circular)
-
-    self.seq_regions : Dict[str, SeqRegion] = defaultdict(seq_region_factory)
-    self.ordered : List[SeqRegion] = field(default_factory=list) # try like this
-    self.syns : Dict[str, str] = dict()
-    self.syn_src_default = syns_src
-    # process
-    #  get actual seq_region names
-    self.fill_length_from_fasta(fasta_file)
-    #  load data from seq_region_raw
-    self.fill_info_from_seq_region_raw(seq_region_raw, use_syns = False)
-    #  get data from meta
-    self.fill_info_from_tech(meta)
-    #  fill synonyms from report file
-    self.fill_info_from_asm_rep(asm_rep_file)
-    #  load from seq_region_genebank, infer names based on the loaded synonyms
-    self.fill_info_from_seq_region_raw(seq_region_genbank, add_missing = False, update_from_new = True, update_syns = False)
-    #  load from seq_region_syns, infer names based on the loaded synonyms
-    self.fill_info_from_seq_region_syns(seq_region_syns)
-
-  def dump(self, json_out: str) -> None:
```

Comment:
don't get this diff

Reply:
2 spaces to 4 spaces tab? So all lines affected

Reply:
it's more of a problem with git. everything is shifted. mb issue with display on my side